spec/app_spec.rb

```diff
@@ -1,13 +1,77 @@
 RSpec.describe App do
-  describe 'visiting /' do
-    it 'responds with 200 to a GET' do
-      get '/'
-      expect(last_response).to be_ok
+  describe 'visiting /authorize' do
+    let(:db) do
+      Sequel.connect(
+        adapter: 'mysql2',
+        host: ENV.fetch('DB_HOSTNAME'),
+        database: ENV.fetch('DB_NAME'),
+        user: ENV.fetch('DB_USER'),
+        password: ENV.fetch('DB_PASS')
+      )
     end
 
-    it 'responds with Hello in the body' do
-      get '/'
-      expect(last_response.body).to eq('Hello')
+    let(:username) { 'abcdef' }
+    let(:client_mac) { 'ee-ee-ee-ee-ee' }
+    let(:ap_mac) { 'ff-ff-ff-ff-ff' }
+    let(:ap_ip_address) { '192.168.1.100' }
+    let(:ap_aruba_name) { 'hallway-1' }
+    let(:ap_meraki_name) { 'hallway-2' }
+
+    after do
+      db[:userdetails].truncate
+    end
+
+    context 'as the Health user' do
+      let(:username) { 'HEALTH' }
+
+      before do
+        db[:userdetails].insert(username: username, password: 'TestUserPassword')
+        get "/authorize/user/#{username}"
```

Comment:
Be good to be clear under what circumstances this shorter URL gets called @afoldesi-gds and then add that context to the test description.

Reply:
This is currently called from health checks from the ELB, with the exact username: HEALTH
We can possibly introduce a change to the RADIUS side to use the shortened version in every case, as at the authoriZation step the extra details are NOT used. (could also change the Z to S)

Reply:
Um, I probably should clarify this. We don't currently use these parameters programmatically in the backend code.
However they contain the mac address and IP of the site which are parameters we can search for as a result of a BAU support request, to see if we received login attempts from a specific site at all, or from a device with a different username (eg. typo in the username)
Removing the long URLs for authorise, we would have to rely on post-auth only, leaving us with difficulties identifying for example an issue with the RADIUS server itself (auth received, but post-auth never sent).

Reply:
Would be good to document that in the README.md @RyanMacG

Reply:
Have raised it as a trello ticket for a future PR